extract_rules/bin/extract_rules.dart

```diff
@@ -151,6 +156,49 @@ $match
     print("    Done writing !");
   }
 
+  // ===== WRITE CHANGELOG.md =====
+  print("[*] Checking git diff...");
+  if (dry) {
+    print("    Dry run, skipped.");
+  } else {
+    var gitDiff = await Process.run(
+      "git",
+      ["diff", "HEAD^", "lib/all.yaml"],
+      workingDirectory: proj.path,
+    );
+
+    final lines = gitDiff.stdout.split("\n");
+    final changes = <String>[];
+    int add = 0;
+    int remove = 0;
+
+    for (var i = 0; i < lines.length; i++) {
+      final line = lines[i];
+      if (line.startsWith("+    - ")) {
+        changes.add("- Added lint rule: `${line.substring(7)}`");
+        add++;
+        print("[i] Added lint rule: ${line.substring(7)}");
+      } else if (line.startsWith("-    - ")) {
+        changes.add("- Removed lint rule: `${line.substring(7)}`");
+        remove++;
```

Comment:
I think we should use a regex to find additions and removals instead. Something like `/([-\+])\s+- (\w+)/` would do the trick and be more flexible.

Reply:
Another option would be to just parse the previous yaml file and the new one and do a diff of the entries in Dart, it might be more robust to future chasnges as well.